assets/pages/regular-contributions/components/regularInlineContributionsPayment.jsx

```diff
@@ -0,0 +1,196 @@
+      props.csrf,
+      props.currency,
+      props.contributionType,
+      props.dispatch,
+      'Stripe',
+      props.referrerAcquisitionData,
+      context.store.getState,
+    )}
+    email={props.email}
+    dispatch={props.dispatch}
+    stripeIsLoaded={props.stripeIsLoaded}
+    isStripeLoaded={props.isStripeLoaded}
+    currency={props.currency}
+    isTestUser={props.isTestUser}
+    isPostDeploymentTestUser={props.isPostDeploymentTestUser}
+    switchStatus={props.stripeSwitchStatus}
+  />);
+
+  let payPalButton = (<PayPalExpressButton
+    amount={props.amount}
+    currency={props.currency}
+    csrf={props.csrf}
+    callback={postCheckout(
+      props.abParticipations,
+      props.amount,
+      props.csrf,
+      props.currency,
+      props.contributionType,
+      props.dispatch,
+      'PayPal',
+      props.referrerAcquisitionData,
+      context.store.getState,
+    )}
+    hasLoaded={props.payPalHasLoaded}
+    setHasLoaded={() => props.dispatch(setPayPalHasLoaded())}
+    switchStatus={props.payPalSwitchStatus}
+  />);
+
+  if (props.hide) {
+    directDebitButton = null;
+    stripeInlineForm = null;
+    payPalButton = null;
+  }
+
+  return (
+    <section className="regular-contribution-payment">
+      { props.paymentStatus === 'Success' ? <Redirect to={{ pathname: routes.recurringContribThankyou }} /> : null }
+      { props.paymentStatus === 'PollingTimedOut' ? <Redirect to={{ pathname: routes.recurringContribPending }} /> : null }
+      {getStatusMessage(props.paymentStatus, props.hide, props.error)}
+      {stripeInlineForm}
+
+      <p className="regular-contribution-payment__or-label">or</p>
+
+      {directDebitButton}
+      {payPalButton}
+    </section>
+  );
+}
+
+// ----- Map State/Props ----- //
+
+function mapStateToProps(state) {
+  return {
+    isTestUser: state.page.user.isTestUser || false,
+    isPostDeploymentTestUser: state.page.user.isPostDeploymentTestUser,
+    email: state.page.user.email,
+    hide: emptyInputField(state.page.user.firstName) || emptyInputField(state.page.user.lastName),
+    error: state.page.regularContrib.error,
+    paymentStatus: state.page.regularContrib.paymentStatus,
+    amount: state.page.regularContrib.amount,
+    currency: state.common.currency,
+    regularContrib: state.page.regularContrib,
+    csrf: state.page.csrf,
+    country: state.common.country,
+    abParticipations: state.common.abParticipations,
+    referrerAcquisitionData: state.common.referrerAcquisitionData,
+    payPalHasLoaded: state.page.regularContrib.payPalHasLoaded,
+    directDebitSwitchStatus: state.common.switches.recurringPaymentMethods.directDebit,
+    stripeSwitchStatus: state.common.switches.recurringPaymentMethods.stripe,
+    payPalSwitchStatus: state.common.switches.recurringPaymentMethods.payPal,
+    stripeIsLoaded: stripeInlineFormActionsFor('regularContributions').stripeIsLoaded,
+    isStripeLoaded: state.page.stripeInlineForm.isStripeLoaded,
+  };
```

Comment:
Can the payment methods be split out into containers? It looks like there's a lot of data being passed into this file and then passed-through into the buttons.